### PR TITLE
chore(C): add fibonacci algorithm

### PR DESCRIPTION
**Pull request by taimoon**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [ ] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] New algorithm
- [x] Optimization in previous algorithms
- [x] Code style update
- [ ] Refactor
- [x] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

An algorithm that compute n-th term Fibonacci number. It includes recursive, iterative, memomized recursive, logarithmic implementation of computing n-th term Fibonacci number .

**Other information:**

### CODE REVIEW

File: algorithms/C/maths/fibonacci-number/Makefile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+run: main.c
+	.\a.exe
+main.c:
+	gcc .\src\main.c
```
Comment (Arsenic-ATG) [inline algorithms/C/maths/fibonacci-number/Makefile:4]:
```suggestion
	gcc .\src\main.c  –lm
```

Not all gcc might be pre-configured to by default link to a proper math library. Explicitly linking the math library make sure the program is built correctly and the can find all the proper definitions that are being used in the program.

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
minor nit-picks

File: algorithms/C/maths/fibonacci-number/src/main.c [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#include<stdio.h>
+#include"fib.c"
```
[the remark this answers is not in the document]
Comment (taimoon) [reply algorithms/C/maths/fibonacci-number/src/main.c]: Yeah, I will do that.

File: algorithms/C/maths/fibonacci-number/Makefile (hunk 1 above)
Comment (taimoon) [reply algorithms/C/maths/fibonacci-number/Makefile]: I will include it.

**Review by Arsenic-ATG — APPROVED** [review]

**Pull request merged**